src/main/java/pl/javastart/task/Main.java

```diff
@@ -1,8 +1,135 @@
 package pl.javastart.task;
 
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAccessor;
+import java.util.*;
+
+import static java.lang.Integer.sum;
+
 public class Main {
 
     public static void main(String[] args) {
+        Main main = new Main();
+        main.run(new Scanner(System.in));
+    }
+
+    public void run(Scanner scanner) {
+        // uzupełnij rozwiązanie. Korzystaj z przekazanego w parametrze scannera
+
+        System.out.println("Podaj datę:");
+        String userInput = scanner.nextLine();
+
+        LocalDateTime localDateTime = null;
+        try {
+            localDateTime = createLocalDateTimeFromDate(userInput);
+        } catch (DateTimeParseException e) {
+            //
+        }
+        if (localDateTime == null) {
+            localDateTime = createLocalDateTimeFromPattern(userInput);
+        }
+        if (localDateTime == null) {
+            localDateTime = createModifiedLocalDateTime(userInput);
+        }
+        if (localDateTime == null) {
+            System.out.println("Nieprawidlowy format daty.");
+        } else {
+            printAllTimeInZones(localDateTime);
+        }
+    }
+
+    private LocalDateTime createModifiedLocalDateTime(String userInput) {
+        LocalDateTime t = LocalDateTime.now();
+
+        List<Character> keys = Arrays.asList('y', 'M', 'd', 'h', 'm', 's');
+        Map<Character, Integer> mapOfAllParameters = new HashMap<>();
+        for (Character key : keys) {
+            mapOfAllParameters.put(key, 0);
+        }
+
+        Map<Integer, Character> mapOfUsedParameters = new HashMap<>();
```

Comment:
Nazwy tych map nie do końca oddają co one właściwie przechowują. Z tego co rozumiem `mapOfAllParameters` to mapa, w której są parametry i przypisane do nich w przyszłości przesunięcia, nazwałbym to więc raczej `parametersWithTimeShift`, albo nawet prościej `timeShifts`.
Z kolei `mapOfUsedParameters`, to mapa, w której kluczami są indeksy, a przypisanymi wartościami "parametry", więc bardziej by pasowało `indicesOfParameters`